File: python/1st_ex/main.py
import math

def err(exc):
    raise Exception(exc)

def init():
    val = int(input())
    if val <= 0:
        err("Invalid input")
    return val

def func(S, N, M):
    if S * M > (S - S // 7) * N:
        return -1
    return math.ceil(S * M / N)

S = init()
N = init()
M = init()

print(func(S, N, M))

File: kotlin/1st_ex/main.kt
import kotlin.math.ceil

fun main() {
    val S = init()
    val N = init()
    val M = init()

    println(func(S, N, M))
}

fun err(exc: String): Nothing {
    throw Exception(exc)
}

fun init(): Int {
    val valStr = readLine() ?: err("Invalid input")
    val value = valStr.toInt()
    if (value <= 0) err("Invalid input")
    return value
}

fun func(S: Int, N: Int, M: Int): Int {
    if (S * M > (S - S / 7) * N) {
        return -1
    }
    return ceil(S * M.toDouble() / N).toInt()
}
